giotto/graphs/create_clique_complex.py

```diff
@@ -0,0 +1,392 @@
+                                "alpha must be a float number.")
+            self.graph = self._create_graph()
+            self.adjacent_matrix = csr_matrix(self.adjacent_matrix)
+
+    def create_complex_from_graph(self):
+        """
+        Functions that computes the Cliques Complex starting from
+        the related networkx graph object.
+
+        Returns
+        -------
+        self.complex_dict : dictionary
+            Dictionary containing all simplices of the clique
+            complex identified with arbitrary ID. Each entry 'K' of the
+            dictionary is again a dictionary contaning all K-simplexes.
+
+        """
+
+        self._find_lists()
+
+        return self.complex_dict
+
+    def _find_lists(self):
+
+        """
+        Collect all cliques of a graph and create dictionary with all simplices
+        """
+
+        # Index for edges and nodes, arbitrary id to edges
+        sim_complex = dict()
+
+        sim_complex[1] = dict(zip(np.arange(
+            nx.number_of_edges(self.graph)), self.graph.edges))
+        sim_complex[0] = dict(zip(self.graph.nodes, self.graph.nodes))
+
+        # Dictionary containing simplexes orders as indexes,
+        # list of tuplas with node id forming the simplexes
+        cliques = list(nx.enumerate_all_cliques(self.graph))
+
+        for x in range(nx.number_of_nodes(self.graph) +
+                       nx.number_of_edges(self.graph), len(cliques)):
+            if sim_complex.get(len(cliques[x]) - 1) is None:
+                i = 0
+                sim_complex[(len(cliques[x]) - 1)] = dict()
+                sim_complex[len(cliques[x]) - 1][i] = tuple(cliques[x])
+                i += 1
+            else:
+                sim_complex[len(cliques[x]) - 1][i] = tuple(cliques[x])
+                i += 1
+
+        self.complex_dict = sim_complex
+
+    def _create_graph(self):
+        distance_to_adjacent = np.vectorize(lambda x:
+                                            1 if x < self.alpha else 0)
+
+        if self.data_type == 'cloud':
+            self.adjacent_matrix = distance_to_adjacent(
+                distance_matrix(self.data, self.data, p=2))
+        else:
+            check_symmetric(self.data)
+            self.adjacent_matrix = distance_to_adjacent(self.data)
+
+        self.adjacent_matrix = self.adjacent_matrix - np.identity(
+            self.data.shape[0])
+        return nx.from_numpy_matrix(self.adjacent_matrix)
+
+    def get_adjacent_matrix(self):
+        return self.adjacent_matrix
+
+    def get_graph(self):
+        return self.graph
+
+    def get_complex_dict(self):
+        return self.complex_dict
+
+
+class CreateBoundaryMatrices(BaseEstimator, TransformerMixin):
+    """Construction of Boundary Matrices from dictionary complex.
+
+    This step computes the boundary matrices that can be used both to
+    analyse the complex and compute the laplacian matrices.
+
+    """
+
+    def __init__(self):
```

Comment:
Generally, if a class does not take any parameters in order to be constructed, is not needed to explicitly write the __init__ function.